src/components/detector.ts

```diff
@@ -88,16 +88,20 @@ export default class Detector extends EventEmitter implements IDetector {
    */
   async enable(idleTimeMs: number = 5000): Promise<void> {
     this._logger.debug('Enabling autozoom persistently', 'Boxfish Detector');
-    let isEnabled = await this.isEnabled();
-    if (isEnabled) return; // already enabled
+
+    const response = this._deviceManager.transport.receiveMessage('detector/az-state', idleTimeMs);
     await this._deviceManager.transport.write('autozoom/enable');
-    // Wait `idleTime` seconds for the camera to load the network and persist the new state
-    await new Promise(resolve => setTimeout(() => resolve(), idleTimeMs));
-    isEnabled = await this.isEnabled();
-    if (!isEnabled) {
-      throw new Error('Autozoom state could not be enabled at the moment!');
+    let isEnabled;
+    try {
+      isEnabled = await response;
+      if (isEnabled) {
```

Comment:
What is the response exactly?

Reply:
response will be whether autozoom is enabled true/false